tests/sentry/workflow_engine/endpoints/test_organization_detector_index.py

```diff
@@ -106,6 +107,39 @@ def test_invalid_project(self):
             status_code=403,
         )
 
+    def test_filter_by_ids(self) -> None:
+        detector = self.create_detector(
+            project_id=self.project.id, name="Test Detector", type=MetricIssue.slug
+        )
+        detector_2 = self.create_detector(
+            project_id=self.project.id, name="Test Detector 2", type=MetricIssue.slug
+        )
+        self.create_detector(
+            project_id=self.project.id, name="Test Detector 3", type=MetricIssue.slug
+        )
+
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params=[("id", str(detector.id)), ("id", str(detector_2.id))],
+        )
+        assert len(response.data) == 2
+        assert {d["id"] for d in response.data} == {str(detector.id), str(detector_2.id)}
+
+        # Test with non-existent ID
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params={"id": "999999"},
+        )
+        assert len(response.data) == 0
+
+        # Test with invalid ID format
+        response = self.get_error_response(
+            self.organization.slug,
+            qs_params={"id": "not-an-id"},
+            status_code=400,
+        )
+        assert "id" in response.data
```

Comment:
It'd be nice to have an assertion on "Invalid ID format" to make this more clear